### PR TITLE
wallet-ext: allow only strong passwords

### PR DESCRIPTION
**Pull request by pchrysochoidis**
* use zxcvbn and allow only passwords with score more than 2

https://user-images.githubusercontent.com/10210143/196224880-3cf6a617-a3b3-438a-b822-52504ba6069a.mov

part of #4926

### CODE REVIEW

**Review by Jibz1 — APPROVED** [review]

**Review by mystie711 — APPROVED** [review]
Really digging the conversational error feedback. 
cc: @randall-Mysten take a look!

**Comment by randall-Mysten** [comment]
> Really digging the conversational error feedback. cc: @randall-Mysten take a look!

Wondering whether it should say "Must contain at least 8 characters" until it has 8 chars, and then "Must contain an uppercase letter and a number" until it does, and then "Password not strong enough" and the repeated words warning. It seems like we want folks to use multi-word-passwords-or-something but don't indicate that until the error message

**Comment by mystie711** [comment]
@randall-Mysten @pchrysochoidis is using a library for all error states that are contextual. So these messages are derived from the set they have. I would not recommend any further edits to the error message so we get all future updates by default.

Library: https://github.com/dropbox/zxcvbn#readme
Error messages: https://github.com/dropbox/zxcvbn/blob/master/src/feedback.coffee

File: apps/wallet/package.json [hunk 1]
```diff
@@ -108,6 +109,7 @@
         "tweetnacl": "^1.0.3",
         "uuid": "^8.3.2",
         "webextension-polyfill": "^0.9.0",
-        "yup": "^0.32.11"
+        "yup": "^0.32.11",
+        "zxcvbn": "^4.4.2"
```
Comment (Jordan-Mysten) [inline apps/wallet/package.json:113]: It's a little scary to me that this hasn't been updated for 7 years.

**Review by Jordan-Mysten — APPROVED** [review]

**Comment by randall-Mysten** [comment]
> @randall-Mysten @pchrysochoidis is using a library for all error states that are contextual. So these messages are derived from the set they have. I would not recommend any further edits to the error message so we get all future updates by default.
> 
> Library: https://github.com/dropbox/zxcvbn#readme Error messages: https://github.com/dropbox/zxcvbn/blob/master/src/feedback.coffee

understood, though this still allows for one to enter a password that is not valid but the error will not indicate specifically why it is not valid.

**Pull request merged**